### PR TITLE
feat(registry-dash): "Add to AI Chat" from Data Exploration

### PR DESCRIPTION
**Pull request by UDtorrey**
## Summary

Lets a registry analyst push the result of an Explore query into the AI chat — either as a fresh chat or as additional context for the existing in-session conversation. Closes the workflow gap where switching from a chart-page modal to **Data Exploration** lost the prior conversation.

- Lifts conversation state from `AiAnalysisModalComponent` into `AiAnalysisService` so it survives modal close and page navigation.
- Adds an **Add to AI Chat** button next to the Run Query controls on the Explore page with a two-item menu: **Start new chat** / **Add to current chat** (the latter disabled when no conversation exists).
- Caps the attached row payload at **100 rows** (`EXPLORE_AI_ROW_CAP`); descriptor + truncation hint included in the user turn.
- Adds a **Start new chat** link to the modal header for explicit reset.
- **Frontend only** — `RegistryDashAiAction` renders `metadata` selectively today, so the descriptor rides in the user-turn text. Structured backend rendering is queued as a follow-up.

## Linear

[SRE-1942](https://linear.app/unstoppable-domains/issue/SRE-1942/featregistry-dash-add-to-ai-chat-from-data-exploration)

## Synergy

Once the parallel `run_explore_query` Tier-3 tool ships (separate PR), Claude can re-run the attached descriptor on demand to fetch more rows than the 100-row cap allows. Mentioned in `metadata.exploreDescriptor` so a future backend follow-up can render it as a structured system-prompt block.

## Follow-up prompts authored to workspace `.context/`

Three standalone prompts for follow-up agents (all depend on this PR):

- `.context/tier3-followup-explore-add-to-chat-persistence-prompt.md` — persist conversation state across page reload.
- `.context/tier3-followup-explore-add-to-chat-multitab-prompt.md` — `BroadcastChannel` cross-tab sync.
- `.context/tier3-followup-explore-add-to-chat-backend-descriptor-prompt.md` — render `metadata.exploreDescriptor` structurally in the system prompt.

## Test plan

- Test 21 added to `.claude/plugins/ud-registry-dash/skills/test-registry-dash/test-plan.md` covering both happy paths and edge cases (button disabled before query runs, "Add to current chat" disabled when no conversation, 100-row truncation).

## Test plan checklist

- [x] `npm run build` passes locally
- [ ] `ng test --browsers=ChromeHeadless --watch=false` — unit specs added; local execution OOM-killed in this environment, will be verified by CI
- [ ] Manual E2E per Test 21 (cross-page resume, new-chat path, edge cases)
- [x] No backend (`core/`) files modified — verified with `git diff master --name-only -- core/` showing only prior-commit files, none in this PR
- [x] Existing sparkle button on Explore page still mounted

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by UDtorrey** [comment]
## CI status

**Passing:** `ud-registry-alpha-gke`, `ud-registry-sandbox-gke`
**Failing:** `ud-registry-alpha`, `ud-registry-crash`, `ud-registry-sandbox`

The 3 non-GKE failures all hit the same pre-existing infrastructure issue at `:console-webapp:buildConsoleWebapp`:

```
Node.js version v22.7.0 detected.
The Angular CLI requires a minimum Node.js version of v20.19 or v22.12.
```

This is **not introduced by this PR** — verified by:
- `git diff master..HEAD -- console-webapp/package.json console-webapp/package-lock.json` is empty (no dep bumps).
- Sibling PR #124 (Tier 3 batch 2) has the exact same failing-non-GKE / passing-GKE pattern on completely unrelated changes.
- The GKE builders pass cleanly on this PR's diff.

The non-GKE Cloud Build trigger images need Node bumped from 22.7.0 to >= 22.12 (or fall back to the v20.19 floor). That's a CI infra fix outside this PR's scope.

**Comment by UDtorrey** [comment]
## Manual E2E results — Test 21 PASS ✅

Ran the full Test 21 plan against a local stack (Java test server :8080 + ng serve :4200, real Anthropic key from Secret Manager). All paths and edge cases pass.

### Stack startup
Note: `start-local-stack.sh` has a `REPO_ROOT` bug (computes `repos/nomulus/.claude` instead of `repos/nomulus` — one `..` short). Ran the test server and ng serve directly to bypass. Filing a separate ticket would be appropriate; not in scope for this PR.

### Path B — Add to current chat (cross-page resume)
1. Domain Activity → sparkle → "Summarize trends" → initial analysis ran.
2. Asked follow-up: *"What's the most common operation type and is that healthy?"* → Claude responded (4-turn conversation, ~3.2KB of dialog text).
3. Closed modal, navigated to Data Exploration.
4. Ran query (`DOMAIN_ACTIVITY` grouped by `tld`, `count_sum`, last 12 months).
5. Clicked **Add to AI Chat** → **Add to current chat**.
6. ✅ Modal reopened with prior conversation visible (head shows the original "Domain Lifecycle Activity Analysis").
7. ✅ New user turn appended: `I just ran an Explore query: DOMAIN_ACTIVITY, grouped by tld, count_sum, last 8760h, month. Descriptor: {...}`.
8. ✅ Claude's response references **both** contexts: *"What about 'example' and 'xn--q9jyb4c' TLDs mentioned earlier?"* (callback to the original conversation) and *"Can you run the same query without the TLD filter?"* (referring to the new Explore query).

### Path A — Start new chat
1. After hard reload (clearing in-memory service state), navigated back to Explore.
2. Ran query.
3. Clicked **Add to AI Chat** → **Start new chat** (after first verifying **Add to current chat** was disabled, see edge case).
4. ✅ Modal opened with title "Explore Analysis — Data Exploration".
5. ✅ No prior conversation, only the explore seed message: *"Summarize the key trends visible in this data."*
6. ✅ Claude analyzed only the Explore data.

### Modal-header "Start new chat" link
1. While in an active chat, clicked the new **Start new chat** link in the modal header.
2. ✅ Prior conversation cleared.
3. ✅ Fresh analysis seeded with explore data.

### Existing sparkle button still resumes after my reset
1. Closed modal mid-conversation, clicked the existing `auto_awesome` sparkle button on the Explore page → "Summarize trends".
2. ✅ Reopened the **same** conversation (same body, "Analysis Summary" content); did not start a new chat.

### Edge cases
- ✅ **"Add to AI Chat" disabled before any query is run** — verified before clicking Run Query (button has `disabled=true`).
- ✅ **"Add to current chat" disabled when no conversation** — verified after hard reload (menu item has `aria-disabled=true`).
- ✅ **Existing sparkle button still mounted on Explore** — both `auto_awesome` (sparkle) and `auto_awesome\nAdd to AI Chat` (new) appear in the button list.

### Network payload (intercepted via `window.fetch` wrapper on the analyze endpoint)
```json
{
  "page": "explore",
  "promptType": "summarize_trends",
  "conversationHistory": [
    {"role": "user", ...},
    {"role": "assistant", ...},
    {"role": "user", "content": "I just ran an Explore query: DOMAIN_ACTIVITY, grouped by tld, count_sum, last 8760h, month.\n\nDescriptor: {\"dataSource\":\"DOMAIN_ACTIVITY\",..."}
  ],
  "metadata": {
    "exploreDescriptor": { "dataSource": "DOMAIN_ACTIVITY", "metrics": [...], "dimensions": [...], "filters": {...}, "granularity": "..." },
    ...
  },
  "chartData": {
    "columns": [...],
    "rows": [...],
    "totalRows": 1,
    "returnedRows": 1,
    "truncated": false
  }
}
```
- `metadata.exploreDescriptor` present and well-formed ✅
- `chartData` carries the truncation wrapper ✅
- (Local fixture only has 14 transactions in 1 TLD, so the 100-row cap couldn't be exercised; the wrapper fields are correct, and the cap is unit-tested in `explore.component.spec.ts`.)

### Console & errors
- 0 errors, 2 (pre-existing) warnings throughout the run.
- No regressions on Domain Activity sparkle or Explore sparkle.

File: console-webapp/src/app/registry-dash/explore/explore.component.ts [hunk 1]
```diff
@@ -95,6 +104,110 @@ export class ExploreComponent implements OnInit {
...
+    const r = this.result();
+    if (!r) return null;
+    const cap = EXPLORE_AI_ROW_CAP;
+    const slicedRows = r.rows.slice(0, cap);
+    return {
+      columns: r.columns,
+      rows: slicedRows,
+      totalRows: r.totalRows,
+      returnedRows: slicedRows.length,
+      truncated: r.rows.length > cap,
```
[the remark this answers is not in the document]
Comment (UDtorrey) [reply console-webapp/src/app/registry-dash/explore/explore.component.ts:117]: This is not on the roadmap for the foreseeable future.

File: console-webapp/src/app/registry-dash/explore/explore.component.ts [hunk 2]
```diff
@@ -95,6 +104,110 @@ export class ExploreComponent implements OnInit {
...
+    const data: AiAnalysisModalData = {
+      title: 'Explore Analysis — Data Exploration',
+      page: 'explore',
+      promptType: 'summarize_trends',
+      userMessage: EXPLORE_PROMPTS[0].userMessage,
+      metadata: this.buildMetadata(),
+      chartData: truncated,
+      isAdmin: false,
+      savedModel: this.savedAiModel(),
+    };
```
[the remark this answers is not in the document]
Comment (UDtorrey) [reply console-webapp/src/app/registry-dash/explore/explore.component.ts:166]: This functionality is not somthing we are considering.

File: console-webapp/src/app/registry-dash/explore/explore.component.ts [hunk 3]
```diff
@@ -109,7 +222,7 @@ export class ExploreComponent implements OnInit {
 
     const header = r.columns.map(escape).join(',');
     const rows = r.rows.map(row => r.columns.map(col => escape(row[col])).join(','));
-    const csv = '\uFEFF' + header + '\n' + rows.join('\n');
+    const csv = '﻿' + header + '\n' + rows.join('\n');
```
[the remark this answers is not in the document]
Comment (UDtorrey) [reply console-webapp/src/app/registry-dash/explore/explore.component.ts:225]: This functionality is not a priority for us.

File: console-webapp/src/app/registry-dash/ai/ai-analysis-modal.component.ts [hunk 4]
```diff
@@ -76,14 +76,15 @@ export class AiAnalysisModalComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.sendInitialRequest();
+    if (!this.aiService.hasActiveConversation()) {
+      this.sendInitialRequest();
+    }
   }
```
[the remark this answers is not in the document]
Comment (UDtorrey) [reply console-webapp/src/app/registry-dash/ai/ai-analysis-modal.component.ts:83]: We are unable to incorporate this into our plans.

File: console-webapp/src/app/registry-dash/ai/ai-analysis.service.ts [hunk 5]
```diff
@@ -28,13 +34,31 @@ export class AiAnalysisService {
   toolsInFlight = signal<ToolInFlight[]>([]);
   toolsUsed = signal<string[]>([]);
 
+  conversationHistory = signal<ConversationMessage[]>([]);
+  lastRequest = signal<LastRequestShape | null>(null);
+  hasActiveConversation = computed(() => this.conversationHistory().length > 0);
```
[the remark this answers is not in the document]
Comment (UDtorrey) [reply console-webapp/src/app/registry-dash/ai/ai-analysis.service.ts:39]: This request is currenly out of scope for the current project phase.

File: console-webapp/src/app/registry-dash/explore/explore.component.html [hunk 6]
```diff
@@ -28,6 +28,27 @@ <h2>Data Exploration</h2>
...
+    </button>
+    <mat-menu #addAiMenu="matMenu">
+      <button mat-menu-item (click)="addToNewChat()">
+        <mat-icon>add</mat-icon>
+        <span>Start new chat</span>
+      </button>
+      <button
+        mat-menu-item
+        (click)="addToCurrentChat()"
+        [disabled]="!aiService.hasActiveConversation()">
```
[the remark this answers is not in the document]
Comment (UDtorrey) [reply console-webapp/src/app/registry-dash/explore/explore.component.html:47]: This proposal does not fit our current needs.
